tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -904,6 +969,39 @@ void HWCommandQueue::enqueue_program(
     this->manager.next_completion_queue_push_back(align(EVENT_PADDED_SIZE, 32), this->id);
 }
 
+void HWCommandQueue::enqueue_record_event(std::reference_wrapper<Event> event) {
+    ZoneScopedN("HWCommandQueue_enqueue_record_event");
+    uint32_t command_size = DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND;
+
+    // Update Event struct for caller
+    auto &event_ref = event.get();
+    event_ref.cq_id = this->id;
+    event_ref.event_id = this->manager.get_next_event(this->id);
+    event_ref.device = this->device;
+
+    if ((this->manager.get_issue_queue_write_ptr(this->id)) + command_size >= this->manager.get_issue_queue_size(this->id)) {
```

Comment:
I think this should be checking that issue_queue_write_ptr + command_size >= issue queue limit

Reply:
Whoops yes I copied buggy version from enqueue-program path, thx for flagging, fixed, about to push.